### PR TITLE
CLN: remove DatetimeLikeArray._add_delta

### PR DESCRIPTION
**Pull request by jbrockmendel**
Ultimately we want to 1) avoid using _from_sequence (so we can make that stricter in what it accepts) and 2) be more careful about when we use `freq="infer"`.  Splitting up the cases handled by _add_delta will make this more feasible.

### CODE REVIEW

File: pandas/core/arrays/datetimelike.py [hunk 1]
```diff
@@ -1151,56 +1151,46 @@ def _sub_period(self, other):
...
+        if new_freq is not None:
+            # fastpath that doesnt require inference
+            return type(self)(new_values, dtype=self.dtype, freq=new_freq)
+        return type(self)._from_sequence(new_values, dtype=self.dtype, freq="infer")
 
     def _add_delta_tdi(self, other):
         """
         Add a delta of a TimedeltaIndex
-        return the i8 result view
+
```
Comment (jreback) [inline pandas/core/arrays/datetimelike.py]: hmm, this name now looks confusing (because of the word delta), maybe

_add_timedeltaindex

**Review by jreback — CHANGES_REQUESTED** [review]
looks good, ex a name

File: pandas/core/arrays/datetimelike.py (hunk 1 above)
Comment (jbrockmendel) [reply pandas/core/arrays/datetimelike.py]: changed to add_timedelta_arraylike

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks. I think could type these functions pretty easily now :->

**Pull request merged**